thirdeye-ui/src/app/rest/datasources/datasources.actions.ts

```diff
@@ -44,3 +50,19 @@ export const useGetDatasources = (): GetDatasources => {
 
     return { datasources: data, getDatasources, status, errorMessages };
 };
+
+// Two unique keys to fetch datasource by, name and id. Hook supports both.
+export const useGetDatasource = (keyType: "name" | "id"): GetDatasource => {
+    const { data, makeRequest, status, errorMessages } =
+        useHTTPAction<Datasource>(
+            keyType === "name" ? getDatasourceByNameREST : getDatasourceREST
+        );
+
+    const getDatasource = (
+        id: string | number
+    ): Promise<Datasource | undefined> => {
+        return makeRequest(id);
+    };
+
+    return { datasource: data, getDatasource, status, errorMessages };
+};
```

Comment:
Please add test cases for this

Reply:
I think I see what you're doing here to reduce duplicate code which is good but sometimes its worth duplicating code for readability. If it were up to me I would just create 2 separate functions called `useGetDatasourceByName` and `useGetDatasourceById` for the sake of simplicity. 

On a separate note do we really need to get a datasource by name if the `id` suffices